src/boosting/gbdt.h

```diff
@@ -179,15 +179,20 @@ class GBDT : public GBDTBase {
       const auto pair = Common::Split(line.c_str(), ":");
       if (pair[1] == " ]")
         continue;
+      const auto param = pair[0].substr(1);
+      const auto value_str = pair[1].substr(1, pair[1].size() - 2);
+      auto iter = param_types.find(param);
+      if (iter == param_types.end()) {
+        Log::Warning("Type for param: '%s' not found. This doesn't affect inference.", param.c_str());
```

Comment:
```suggestion
        Log::Warning("Ignoring unrecognized parameter '%s' found in model string.", param.c_str());
```

I understand that an error about types is the first place where an error is thrown in this situation... but from a user's perspective, I don't think that's important and in fact might be confusing.

I also don't think "This doesn't affect inference" is necessary. And it might become incorrect if parameters that do affect scoring trigger this warning in the future.

Would you consider this re-write? I think it's sufficient just to say "hey LightGBM found this parameter and skipped it".